packages/gatsby-source-nacelle/src/source-nodes/content.js

```diff
@@ -0,0 +1,83 @@
+const { nacelleClient } = require('../services');
+const {
+  cmsPreviewEnabled,
+  replaceKey,
+  cacheIsInvalid,
+  hasBeenIndexedSinceLastBuild
+} = require('../utils');
+
+module.exports = async function (gatsbyApi, pluginOptions) {
+  const { actions, createContentDigest, cache } = gatsbyApi;
+  const { createNode, touchNode } = actions;
+
+  const {
+    nacelleSpaceId,
+    nacelleGraphqlToken,
+    contentfulPreviewSpaceId,
+    contentfulPreviewApiToken
+  } = pluginOptions;
+
+  const client = nacelleClient({
+    previewMode: cmsPreviewEnabled(pluginOptions),
+    nacelleSpaceId,
+    nacelleGraphqlToken,
+    contentfulPreviewSpaceId,
+    contentfulPreviewApiToken
+  });
+
+  try {
```

Comment:
This looks very similar to `collections`. Is there an opportunity to abstract some of this into small functions that can be used across both data types?

Reply:
UPDATE: refactored to use IoC / DI to stay DRY